### PR TITLE
add circleci config / goreleaser to help to cut a release 

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
- add circleci config
- add goreleaser to help in the release process

TODO:

- add gitlab that will run internally when we tag to release

#### Ticket Link
n/A

### CODE REVIEW

**Comment by cpanato** [comment]
tagging @jespino @isacikgoz @streamer45 to get some early feedback

**Comment by cpanato** [comment]
the `new` branch is the correct one? maybe can we rename that to `main` WDTY?

**Comment by streamer45** [comment]
> the `new` branch is the correct one? maybe can we rename that to `main` WDTY?

It was added to keep support on older server branches (prior to 6.0). More context at https://community-daily.mattermost.com/private-core/pl/g5txdxohhbdgmnhuwpfod6cwge

**Comment by streamer45** [comment]
@cpanato Do you need help from us to fix tests/linting failures?

**Comment by cpanato** [comment]
> @cpanato Do you need help from us to fix tests/linting failures?

no, I will do that,

**Comment by cpanato** [comment]
ready @streamer45

**Review by streamer45 — APPROVED** [review]
Looks great :+1:

File: go.mod [hunk 1]
```diff
@@ -1,10 +1,18 @@
 module github.com/mattermost/mattermost-govet/v2
 
-go 1.13
+go 1.17
```
Comment (agnivade) [inline go.mod]: Can we keep this as 1.13?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    steps:
+      - checkout
+      - run:
+          name: Run snapshot
+          command: make snapshot
+
+  test:
+    docker:
+      - image: cimg/go:1.17-node
+    working_directory: ~/go/src/github.com/mattermost/mattermost-govet
```
Comment (agnivade) [inline .circleci/config.yml]: This can be just `~/mattermost-govet`

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            pushd mattermost-api-reference
+            make build
+            popd
+            popd
+      - run:
+          name: Run tests
+          command: |
+            ls -la
+            pwd
+            ls -la ../
```
Comment (agnivade) [inline .circleci/config.yml]: Wondering if we can clean these up?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            mkdir -p /tmp/test-reports
+            gotestsum --junitfile /tmp/test-reports/unit-tests.xml
```
Comment (agnivade) [inline .circleci/config.yml]: Can we just call `make test` here? We are not really using Java, so not sure if there's a benefit of using a junitfile. It's better to just call the standard Go toolchain.

**Review by agnivade — COMMENTED** [review]
Thanks Carlos. Just some small suggestions.

File: .circleci/config.yml (hunk 3 above)
Comment (cpanato) [reply .circleci/config.yml]: this is left overs 😅  that i forgot to remove in the rebase thanks

File: .circleci/config.yml (hunk 4 above)
Comment (cpanato) [reply .circleci/config.yml]: the benefit of this is that generate the circleci report when tests fail, but I can remove

File: go.mod (hunk 1 above)
Comment (cpanato) [reply go.mod]: any particular reason?

File: .circleci/config.yml (hunk 4 above)
Comment (cpanato) [reply .circleci/config.yml]: removed

File: go.mod (hunk 1 above)
Comment (agnivade) [reply go.mod]: Bumping up to 1.17 means the minimum required version to run the project becomes 1.17. Unless there is a dependency on 1.17, I'd prefer to keep it to 1.13.

Comment (cpanato) [reply go.mod]: ok, thanks for clarify

File: .circleci/config.yml (hunk 4 above)
Comment (agnivade) [reply .circleci/config.yml]: Ahh yes, now I remember. And I see that in the documentation for CircleCI as well. It's a bit of a weird restriction from CircleCI's end as they are forcing a different test output format. So I am fine if you want to bring this back.

**Review by agnivade — APPROVED** [review]
Thanks Carlos. Up to you if you want to bring back the junitfile. 0/5.

File: .circleci/config.yml (hunk 4 above)
Comment (cpanato) [reply .circleci/config.yml]: I'm ok to keep it as is, I'm lazy now to revert :P

**Review by isacikgoz — APPROVED** [review]
LGTM! This is awesome @cpanato 🎉

**Review by jespino — APPROVED** [review]
LGTM

**Pull request merged**